src/backend/distributed/executor/partitioned_intermediate_results.c

```diff
@@ -179,6 +180,23 @@ worker_partition_query_result(PG_FUNCTION_ARGS)
 						errmsg("query must generate a set of rows")));
 	}
 
+	if (columnCount != -1)
+	{
+		if (columnCount < 1 || columnCount > tupleDescriptor->natts)
+		{
+			ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+							errmsg("column count must be between 1 and %d (%d)",
+								   tupleDescriptor->natts, columnCount)));
+		}
+
+		/*
+		 * Caller of this function might request to reduce number of output
+		 * columns, because they don't want to get the target entries added
+		 * in ExpandWorkerTargetEntry() in the output.
```

Comment:
Are there any alternative to solving it here?

Reply:
> Are there any alternatives to solving it in this function?

The best way would be refactor planner to remove extra columns if there is no coordinator work to do. I'll work on that today.

Other alternative is wrapping the task query and selecting only columns we need, etc. might be a little brittle as I need to make assumptions about how the planner assigns names, etc.

Reply:
Wrapped task query to select only columns we are interested in. I wasn't aware of `job->jobQuery` which can be used to get list of columns assigned by citus planner.